zbar_lite/setup.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# _*_coding:utf-8_*_
+"""
+@Time   :  2021/2/7 18:51
+@Author :  wqh
+@Email  :  597935261@qq.com
+"""
+try:
+    import setuptools
+    from setuptools import setup, Extension
+    from setuptools.command.install import install
+    from setuptools.command.build_ext import build_ext
+
+
+    class MyBuildExt(build_ext):
+        """
+        MyBuildExt
+        """
+
+        def build_extensions(self):
+            """
+
+            :return:
+            """
+            self.compiler.dll_libraries = []
+            build_ext.build_extensions(self)
+
+
+    setuptools_opts = dict(cmdclass={"install": install, "build_ext": MyBuildExt})
+
+except ImportError as err:
+    print(err)
+    from distutils.core import setup
+    from distutils.extension import Extension
+
+    setuptools_opts = {}
+import os
+import ctypes
+import ctypes.util
+
+CUR_DIR = os.path.realpath(os.path.dirname(__file__))
+
+
+def get_c_file(target_dir):
+    """
+    to find the .c file
+    :param target_dir:
+    :return:
+    """
+    return [os.path.join(target_dir, x) for x in os.listdir(target_dir) if x.endswith(".c")]
+
+
+SRCS = get_c_file("src/zbar") + \
+       get_c_file("src/zbar/decoder") + \
+       get_c_file("src/zbar/video") + \
+       get_c_file("src/zbar/qrcode") + \
+       get_c_file("src/zbar/processor") + \
+       get_c_file("src/zbar/window") + \
+       get_c_file("zbar")
+
```

Comment:
No, the "src" is created by preparation.sh which is used to copy what i need to a working dir.

Reply:
Also there will be no DESTDIR for pythonic setup, it will be installed in the PYTHON PATH in your env.

Reply:
> No, the "src" is created by preparation.sh which is used to copy what i need to a working dir.

Instead of having a separate preparation.sh, please handle everything that will be needed via `configure.ac` (without copying files to a separate place)

> Also there will be no DESTDIR for pythonic setup, it will be installed in the PYTHON PATH in your env.

No, you still need to honour DESTDIR, as depending on the compilation options, and the OS, it may be generating files either under a chroot environment or setting a destdir to some place, in order to generate an install package.

See, `configure.ac` deals with different scenarios:

1. when an user compile a package, it installs directly under $prefix (usually `/usr`, '/opt' or `/usr/local` - depending on the user's preference)
2. when a distro packager produces a package, this usually installed at `<some path>/usr`

This is valid *even* for python packages.

If you want to see scenario (2), try this (no need to be `root`):
```
DESTDIR=install make install
```
This will make a new directory called `./install` and will place all binaries there. A poor man's binary package could be generated with:

```
(DESTDIR=${PWD}/install make install && cd install && zip ../zbar-binaries.zip *)
```